chain/chaintest/block.go

```diff
@@ -42,3 +64,271 @@ func GenerateEmptyExecutedBlocks(
 	}
 	return executedBlocks
 }
+
+// TxListGenerator is used to populate blocks for benchmarking.
+type TxListGenerator interface {
+	// Generate returns a list of transactions that will be the TX list of a
+	// block (can be empty or can have a length not equal to [numOfTxs]).
+	//
+	// [factories] is a list of unique accounts whose length is equal to [numOfTxs].
+	Generate(
+		rules chain.Rules,
+		timestamp int64,
+		unitPrices fees.Dimensions,
+		factories []chain.AuthFactory,
+		numOfTxs uint64,
+	) ([]*chain.Transaction, error)
+}
+
+// NoopTxListGenerator is a TxListGenerator that generates an empty TX list
+type NoopTxListGenerator struct{}
+
+func (NoopTxListGenerator) Generate(
+	chain.Rules,
+	int64,
+	fees.Dimensions,
+	[]chain.AuthFactory,
+	uint64,
+) ([]*chain.Transaction, error) {
+	return []*chain.Transaction{}, nil
+}
+
+// GenerateExecutionBlocks generates [numBlocks] execution blocks with
+// [txsPerBlock] transactions per block.
+//
+// Block production is a simplified version of Builder; we execute
+// transactions followed by writing the chain metadata to the state diff.
+func GenerateExecutionBlocks(
+	ctx context.Context,
+	rules chain.Rules,
+	metadataManager chain.MetadataManager,
+	balanceHandler chain.BalanceHandler,
+	txListGenerator TxListGenerator,
+	factories []chain.AuthFactory,
+	parentView state.View,
+	parentID ids.ID,
+	parentHeight uint64,
+	parentTimestamp int64,
+	numBlocks uint64,
+	numOfTxsPerBlock uint64,
+) ([]*chain.ExecutionBlock, error) {
+	var timestampOffset int64
+	switch numOfTxsPerBlock {
+	case 0:
+		timestampOffset = rules.GetMinEmptyBlockGap()
+	default:
+		timestampOffset = rules.GetMinBlockGap()
+	}
```

Comment:
Can we handle this with a single if block instead of a switch? 

Not identical, but this is a case of "unnecessary else" - https://github.com/uber-go/guide/blob/master/style.md#unnecessary-else